### PR TITLE
cpuid: Unify implementations for the simplest cases

### PR DESCRIPTION
**Pull request by gebart**
Most of the CPUs which provide device unique IDs have an implementation which only consists of a memcpy from some hardware register to the address specified. This PR unifies those platforms into one implementation.

Also deleted the lm4f120 cpuid implementation because it was simply not providing the intended functionality. The CPU ID on that platform was wrongfully using the Cortex-M family ID, which is the same for all Cortex-M cores of a particular family and patch level, even across different vendors.

Update: cc2538 based platforms will get new CPU IDs with this PR, there was a bug in the original driver which created a CPU ID which was not the same as the hardware register. Also, the CPU ID on cc2538 may be byte-swapped compared to the EUI64 of the radio.

### CODE REVIEW

File: cpu/sam0_common/periph/cpuid.c [hunk 1]
```diff
@@ -33,5 +33,5 @@
 void cpuid_get(void *id)
 {
     uint32_t addr[] = { WORD0, WORD1, WORD2, WORD3 };
-    memcpy(id, (void *)addr, CPUID_LEN);
+    memcpy(id, &addr[0], CPUID_LEN);
```
Comment (miri64) [inline cpu/sam0_common/periph/cpuid.c]: in theory, even `memcpy(id, addr, CPUID_LEN)` should suffice.

**Comment by gebart** [comment]
@miri64 sorry, forgot to `git add`, amended

File: cpu/cc2538/periph/cpuid.c [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-
-void cpuid_get(void *id)
-{
-    uint8_t *dest = (uint8_t *)id;
-
-    /**
-     * The byte-order is big-endian but the word order is little endian.
-     * Make some sense of it:
-     */
-    dest[0] = IEEE_ADDR_MSWORD >> (3 * sizeof(uint8_t));
```
Comment (gebart) [inline cpu/cc2538/periph/cpuid.c]: I'm fairly convinced this was completely wrong in the original implementation, the sizeof() is in bytes, but the shift should be in number of bits.

**Comment by gebart** [comment]
Added note on cc2538 byte swap in the top post.

**Pull request merged**